### PR TITLE
pvs/V1028 screen.c:6083

### PR DESCRIPTION
**Pull request by ngortheone**
Fixing instances of V1028 from PVS report labeled as level high, one at a time

As per @justinmk request here https://github.com/neovim/neovim/pull/10478

### CODE REVIEW

File: src/nvim/screen.c [hunk 1]
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
 {
   int new_row;
   ScreenGrid new = *grid;
-
-  size_t ncells = (size_t)((rows+1) * columns);
+  /* Since amount of cells could not be negative we expect both
```
Comment (bfredl) [inline src/nvim/screen.c]: use `//` comments.

File: src/nvim/screen.c [hunk 2]
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
...
+   * rows and columns to be positive */
+  assert(rows >= 0 && columns >= 0);
+  size_t ncells = ((size_t)rows + 1) * columns;
```
Comment (bfredl) [inline src/nvim/screen.c]: the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

Comment (ngortheone) [reply src/nvim/screen.c]: done

File: src/nvim/screen.c [hunk 3]
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
 {
   int new_row;
   ScreenGrid new = *grid;
-
-  size_t ncells = (size_t)((rows+1) * columns);
+  // Since amount of cells could not be negative we expect both
+  // rows and columns to be positive
+  assert(rows >= 0 && columns >= 0);
+  size_t ncells = (size_t)rows  * columns;
```
Comment (justinmk) [inline src/nvim/screen.c]: too many spaces.

File: src/nvim/screen.c (hunk 2 above)
Comment (justinmk) [reply src/nvim/screen.c]: > the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

that explanation should be pasted into the commit message.

**Comment by justinmk** [comment]
> the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

that explanation should be pasted into the commit message.

**Comment by ngortheone** [comment]
> that explanation should be pasted into the commit message.

@justinmk done

File: src/nvim/screen.c (hunk 3 above)
Comment (ngortheone) [reply src/nvim/screen.c]: fixed

File: src/nvim/screen.c [hunk 4]
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
 {
   int new_row;
   ScreenGrid new = *grid;
-
-  size_t ncells = (size_t)((rows+1) * columns);
+  // Since amount of cells could not be negative we expect both
+  // rows and columns to be positive
```
Comment (justinmk) [inline src/nvim/screen.c]: the comment is redundant with the  `assert()`

Comment (ngortheone) [reply src/nvim/screen.c]: removed

**Comment by justinmk** [comment]
Thanks. For PVS commit messages I usually include some form of the PVS warning or advice. For V1028 that is "cast operands, not the result".

**Comment by ngortheone** [comment]
ack. will add to other commit messages.

**Pull request merged**